oteapi/strategies/parse/image.py

```diff
@@ -93,53 +143,85 @@ def get(
         session.update(downloader.initialize(session))
-        cache_key = downloader.get(session).get("key", "")
 
-        cache = DataCache(self.parse_config.configuration.datacache_config)
+        downloader = create_strategy("download", download_config)
+        output = downloader.get(session)
+        session.update(output)
+
+        if config.datacache_config and config.datacache_config.accessKey:
+            cache_key = config.datacache_config.accessKey
+        elif "key" in output:
+            cache_key = output["key"]
+        else:
+            RuntimeError("No data cache key provided to the downloaded content")
+
+        cache = DataCache(config.datacache_config)
 
         # Treat image according to filter values
         with cache.getfile(cache_key, suffix=mime_format) as filename:
-            image = Image.open(filename, formats=[image_format]).crop(
-                self.parse_config.configuration.crop
+            image = Image.open(filename, formats=[image_format])
+            if crop:
+                image = image.crop(crop)
+            if config.image_mode:
+                image = image.convert(mode=config.image_mode)
+
+            if image_format == "GIF":
+                if image.info.get("version", b"").startswith(b"GIF"):
+                    image.info.update(
+                        {"version": image.info.get("version", b"")[len(b"GIF") :]}
+                    )
+
+            # Use the buffer protocol to store the image in the datacache
+            data = np.asarray(image)
+            image_key = cache.add(
+                data,
+                key=config.image_key,
+                tag=str(id(session)),
             )
 
-        if image_format == "GIF":
-            if image.info.get("version", b"").startswith(b"GIF"):
-                image.info.update(
-                    {"version": image.info.get("version", b"")[len(b"GIF") :]}
+            if image.mode == "P":
+                image_palette_key = cache.add(
+                    np.asarray(image.getpalette()), tag=str(id(session))
                 )
+            else:
+                image_palette_key = None
+
+            # The session must be json serialisable - filter out all
+            # non-json serialisable fields in image.info
+            if image.info:
+                image_info = {
+                    key: val
+                    for key, val in image.info.items()
+                    if isinstance(val, (str, int, float, type(None), bool, tuple, list))
+                }
+            else:
+                image_info = {}
+
+            session_update = SessionUpdateImageParse(
+                image_key=image_key,
+                image_size=image.size,
+                image_mode=image.mode,
+                image_palette_key=image_palette_key,
+                image_info=image_info,
+            )
+
+            # Explicitly close the image to avoid crashes on Windows
+            image.close()
```

Comment:
Why not simply return the generated `SessionUpdateImageParse` straight up, instead of storing it in an unused variable first?
```suggestion
            # Explicitly close the image to avoid crashes on Windows
            image.close()

            return SessionUpdateImageParse(
                image_key=image_key,
                image_size=image.size,
                image_mode=image.mode,
                image_palette_key=image_palette_key,
                image_info=image_info,
            )
```

Reply:
Because we are using the `image` object for creating the `session_update`. But the image must be closed to avoid Windows failing to remove the temporary file as mentioned in the comment.